### PR TITLE
Issue 12 - Handle deleted translations

### LINKED ISSUE #12: Correctly handle translations which have been deleted from en.yaml files.
### Description
If a latest translation already exists, but doesn't exist in latest sync, then it should not be served and should not show in latest sync.

Currently this is an issue, if a translation has been removed from the repo's yaml, then the repos translations are synced, and the locale files are generated the deleted yaml will be included.

Open questions:
* How will this work with english vs other langauges?
  * It should work the same with all langauges but will need to see.
* Need to handle an entire yaml file being deleted!!! Currently the deleted file will just be written.
  * This would be good to write tests for. 
 
### Requirements
- A field should be added to the Translation model called `deleted_at` which is a DateTime field. It should be able to be blank and editable.
- When a sync is made, check if any of the results of `filter_latest_translations`did not exist in the latest sync. If it doesn't exist, then set the `deleted_at` value to the commit time.
- When the latest translation has a value in the `deleted_at` field it should not be exported when generating the locale files.

### PR DESCRIPTION
**Pull request by djbusstop**
Resolves #12 
 
This PR:
* Adds `deleted_at` to Translation object
* `deleted_at` is set on Translations when they are removed from projects, identified in `sync_project`
  * Creates new method `mark_deleted_translations` which has test
* Any translation that has `deleted_at` set will be excluded from the generated locale files output
  * This is tested 
* `filter_latest_translations()` is modified to only return the non-deleted Translations by default
* Changes admin interface:
  * Link on project rows, navigates to all translations for that project.
  * Shows if a translation has been deleted on a translation row.
  * Readonly fields for created and deleted time for translation
* `/translations/` endpoint only returns translations that have **not** been deleted
  * Therefore, editors can only edit translations that haven't been deleted

### CODE REVIEW

**Comment by djbusstop** [comment]
Do we want to add "deleted at" to every previous one?"
  - Yes we do need to add deleted at on all

Why are the root pages not being added?
  - Deleted at is being added to translations that are not actually deleted

Need to exclude deleted from frontend response

**Comment by djbusstop** [comment]
This work is complicated by the fact translations are not related to one another. Currently, the delete function works by getting all translations at a file path and object path, and then adding "deleted_at" to all of them. Should we rather allow only one to have it set, and have a hook which edits all the other ones as being deleted? Not sure....

File: .env.development-sample [hunk 1]
```diff
@@ -1 +1,4 @@
 GITHUB_ACCESS_TOKEN=
+DJANGO_SECRET_KEY=secret_key
+DEBUG=1
+ENVIRONMENT=development
```
Comment (djbusstop) [inline .env.development-sample:4]: Required for local testing if someone wants to do that.

**Comment by djbusstop** [comment]
Todo:
- [] Create function for handling deleted translations and test it
- [] Possibly remove test for deleted translations on sync

**Review by richardolsson — APPROVED** [review]
Looking good! No objections from me. 🙌

**Pull request merged**